joomla/joomla-cms
# [4.0] Pass $app to Module Helper

Pull Request for Issue # .

### Summary of Changes
This PR modify some of our frontend modules code to pass application option (which is available via $app variable) to module helper methods. This helps:

- Using injected $app object instead of having to call Factory::getApplication() to get application object (isn't it the purpose of our new module dispatcher architecture?) ?
- Allow replacing deprecated Factory::getUser() method with $app->getIdentity();

As these helper methods change in this PR is public method, I'm unsure if changing parameters here are consider backward incompatible changes. If it is, please let me know and I will close the PR. Just want to improve the code a bit.


### Testing Instructions
- Code review
- Or publish all the modules changed in this PR and make sure it is still working as expected.
